pkg/scheduler/preemption/preemption.go

```diff
@@ -63,14 +64,35 @@ type Preemptor struct {
 	applyPreemption func(ctx context.Context, w *kueue.Workload, reason, message string) error
 }
 
+type FlavorResourceSet struct {
```

Comment:
why not just `type FlavorResourceSet sets.Set[resources.FlavorResource]`

Reply:
I tried this at first, but then it's not possible to call methods of `sets.Set` directly from `FlavorResourceSet`